### ISSUE TITLE
rake aborted when converting  Journal

### ISSUE THREAD

**Issue by hollychen503** [Ecodev/redmine_convert_textile_to_markown#1]
pandoc 1.18
redmine 3.3.0
ruby-2.3.1

```
...pandoc: Prelude.head: empty list
rake aborted!
pandoc failed
/var/www/html/redmine3/lib/tasks/convert_textile_to_markdown.rake:86:in `convert_textile_to_markdown'
/var/www/html/redmine3/lib/tasks/convert_textile_to_markdown.rake:21:in `block (4 levels) in <top (required)>'
/var/www/html/redmine3/lib/tasks/convert_textile_to_markdown.rake:17:in `each'
/var/www/html/redmine3/lib/tasks/convert_textile_to_markdown.rake:17:in `block (3 levels) in <top (required)>'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/activerecord-4.2.6/lib/active_record/relation/batches.rb:51:in `block (2 levels) in find_each'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/activerecord-4.2.6/lib/active_record/relation/batches.rb:51:in `each'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/activerecord-4.2.6/lib/active_record/relation/batches.rb:51:in `block in find_each'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/activerecord-4.2.6/lib/active_record/relation/batches.rb:124:in `find_in_batches'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/activerecord-4.2.6/lib/active_record/relation/batches.rb:50:in `find_each'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/activerecord-4.2.6/lib/active_record/querying.rb:9:in `find_each'
/var/www/html/redmine3/lib/tasks/convert_textile_to_markdown.rake:16:in `block (2 levels) in <top (required)>'
/var/www/html/redmine3/lib/tasks/convert_textile_to_markdown.rake:14:in `each'
/var/www/html/redmine3/lib/tasks/convert_textile_to_markdown.rake:14:in `block in <top (required)>'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/task.rb:248:in `block in execute'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/task.rb:243:in `each'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/task.rb:243:in `execute'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/task.rb:187:in `block in invoke_with_call_chain'
/home/ubuntu/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/task.rb:180:in `invoke_with_call_chain'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/task.rb:173:in `invoke'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/application.rb:152:in `invoke_task'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/application.rb:108:in `block (2 levels) in top_level'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/application.rb:108:in `each'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/application.rb:108:in `block in top_level'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/application.rb:117:in `run_with_threads'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/application.rb:102:in `top_level'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/application.rb:80:in `block in run'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/application.rb:178:in `standard_exception_handling'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/lib/rake/application.rb:77:in `run'
/home/ubuntu/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/home/ubuntu/.rvm/gems/ruby-2.3.1/bin/rake:23:in `load'
/home/ubuntu/.rvm/gems/ruby-2.3.1/bin/rake:23:in `<main>'
/home/ubuntu/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/home/ubuntu/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => convert_textile_to_markdown
```

**Comment by PowerKiKi** [comment]
This is most likely an issue with pandoc itself, you should file a issue on [their project](https://github.com/jgm/pandoc) with the detail of the textile source and the exact command which was used. You can dig out those information by debugging before the `system` call, by adding `p textile` and `p command`.

But it could also be a custom syntax valid in Redmine, but invalid in the official Textile supported by pandoc. You'll have to find out by yourself...

**Comment by hollychen503** [comment]
I know nothing about ruby and pandoc.
Just add some code to ignore the exception.
It's not a good solution but just work for me .

```
          begin
            markdown = convert_textile_to_markdown(textile)
            model.update_column(attribute, markdown)
          rescue    
            puts 'rescued'  
          end
```

**Comment by universam1** [comment]
same here, thanks for the good hint with the rescue block!

**Comment by universam1** [comment]
@PowerKiKi Great work! Any reason not to use the rescue block?

**Comment by PowerKiKi** [comment]
yes, there is a reason. The failure means pandoc crashed. If pandoc crashed, then your content could not be migrated to Markdown. if this happens often, then you are not migrating your content at all anymore. It is **much** better to analyze what the original textile was, and submit a bug report to the pandoc project. That way they will be able to improve pandoc for everyone, and you will be able to migrate more of your content automatically and flawlessly.

tl;dr: it's lazy and counter-productive
